src/MongoDB/ClientEncryption.c

```diff
@@ -70,6 +151,44 @@ static PHP_METHOD(ClientEncryption, createDataKey)
 	phongo_clientencryption_create_datakey(intern, return_value, kms_provider, options);
 } /* }}} */
 
+/* {{{ proto object MongoDB\Driver\ClientEncryption::deleteKey(MongoDB\BSON\Binary $id)
+   Removes the key document with the given UUID (BSON binary subtype 0x04) from
+   the key vault collection. Returns the result of the internal deleteOne()
+   operation on the key vault collection. */
+static PHP_METHOD(ClientEncryption, deleteKey)
+{
+	zval*        zkeyid = NULL;
+	bson_value_t keyid  = { 0 };
+	bson_t       reply  = BSON_INITIALIZER;
+	bson_error_t error  = { 0 };
+
+	PHONGO_PARSE_PARAMETERS_START(1, 1)
+	Z_PARAM_OBJECT_OF_CLASS(zkeyid, php_phongo_binary_ce)
+	PHONGO_PARSE_PARAMETERS_END();
+
+	php_phongo_zval_to_bson_value(zkeyid, PHONGO_BSON_NONE, &keyid);
+
+	if (EG(exception)) {
+		goto cleanup;
+	}
+
+	if (!validate_keyid(&keyid)) {
+		/* Exception already thrown */
+		goto cleanup;
+	}
+
+	if (!mongoc_client_encryption_delete_key(Z_CLIENTENCRYPTION_OBJ_P(getThis())->client_encryption, &keyid, &reply, &error)) {
+		phongo_throw_exception_from_bson_error_t(&error);
+		goto cleanup;
+	}
+
+	RETVAL_OPTIONAL_BSON_T(reply);
```

Comment:
Strictly speaking, this could return `null` if `reply` were to come back empty for some reason, which would then cause a fatal error due to a return type mismatch. In my opinion we should handle that before to ensure we can't run into the `RETVAL_NULL()` case without throwing an exception. Alternatively, we can change the return value to `object|null` if that makes sense.

Reply:
Good point. I believe this is the only method where the reply document should never be empty. Other methods allow empty documents when nothing matches.

I'll add a catch for this that throws an UnexpectedValueException before we call `RETVAL_OPTIONAL_BSON_T`.